File: sentimental_mario_less/mario.py
# Print a pyramid of maximum height of 23 based on the users input height

from cs50 import get_int

# Maximum height of the pyramid
HEIGHT = 23

# If number is negative or greater than 23, reprompt the user for input
while True:
    n = get_int("Height of pyramid less than 23: ")
    if n >= 0 and n < (HEIGHT + 1):
        break

# Print columns
for i in range(n):

    # Print spaces in rows
    for j in range(n - (i + 1)):
        print(" ", end="")

    # Print bricks in rows
    for k in range(i + 1):
        print("#", end="")

    # Print additional column
    print("#")

File: mario_less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Ask user for height n that is a positive number less than 23
    int n;
    int j;
    int k;
    do
    {
        n = get_int("Height of pyramid less than 23: ");
    }
    while (n < 0 || n > 23);

    // Print height of pyramid
    for (int i = 0; i < n; i++)
    {
        // Print spaces
        for (k = (i + 1); k < n; k++)
        {
            printf(" ");
        }

        // Print hashes
        for (j = (n - (i + 1)); j < n; j++)
        {
            printf("#");
        }

        // Print additional column
        printf("#");

        // Print new line
        printf("\n");
    }
}
